packages/playwright-test/src/runner.ts

```diff
@@ -270,13 +271,41 @@ export class Runner {
       });
       filesByProject.set(project, testFiles);
     }
-    return { filesByProject, setupFiles };
+
+    let applyFilterToSetup = false;
+    if (testFileFilters.length && setupFiles.size) {
+      // If the filter didn't match any tests apply it to the setup files.
+      if (setupFiles.size === fileToProjectName.size) {
+        applyFilterToSetup = true;
+        for (const [project, files] of filesByProject) {
+          const filteredFiles = files.filter(testFileFilter);
+          if (filteredFiles.length)
+            filesByProject.set(project, filteredFiles);
+          else
+            filesByProject.delete(project);
+        }
+        for (const file of setupFiles) {
+          if (!testFileFilter(file))
+            setupFiles.delete(file);
+        }
+      } else {
+        const setupFile = [...setupFiles].find(testFileFilter);
+        // If the filter is not empty and it matches both setup and tests then it's an error: we allow
+        // to run either subset of tests with full setup or partial setup without any tests.
+        if (setupFile) {
+          const testFile = Array.from(fileToProjectName.keys()).find(f => !setupFiles.has(f));
+          throw new Error(`Both setup and test files match command line filter.\n  Setup file: ${setupFile}\n  Test file: ${testFile}`);
```

Comment:
- Let's print relative to `config.rootDir`. 
- We can also print the number of matched files if more than one.